### PR TITLE
[Anchor] Use STF and ETF

### PR DESCRIPTION
**Pull request by benedikt-voelkel**
Use BasicCCDBManager's getRunDuration to determine time range to be covered by simulation.

This will first try to get STF and ETF. If not available, the fallback chain is
1. SOX and EOX
1. SOR and EOR

### CODE REVIEW

File: MC/bin/o2dpg_sim_workflow_anchored.py [hunk 1]
```diff
@@ -150,22 +137,18 @@ def retrieve_params_fromGRPECS(ccdbreader, run_number, rct = None):
...
-       # verify that the variaous sor_eor information are the same
-       if SOR != rct["SOR"]:
-         print ("WARNING: Inconsistent SOR information on CCDB (divergence between GRPECS and RCT) ... will take RCT one")
-         SOR=rct["SOR"]
-
-       if EOR != rct["EOR"]:
-         print ("WARNING: Inconsistent EOR information on CCDB (divergence between GRPECS and RCT) ... will take RCT one")
-         EOR=rct["EOR"]
+    if run_start is not None:
+       print (f"INFO: GRPECS SOR ({SOR}) will be superseded by externally provided run start ({run_start})")
```
Comment (chiarazampolli) [inline MC/bin/o2dpg_sim_workflow_anchored.py:147]: This is not anymore from GRPECS, right?

File: MC/bin/o2dpg_sim_workflow_anchored.py [hunk 2]
```diff
@@ -150,22 +137,18 @@ def retrieve_params_fromGRPECS(ccdbreader, run_number, rct = None):
...
+       SOR = run_start
+    if run_end is not None:
+       print (f"INFO: GRPECS EOR ({EOR}) will be superseded by externally provided run end ({run_end})")
```
Comment (chiarazampolli) [inline MC/bin/o2dpg_sim_workflow_anchored.py:150]: As above.

File: MC/bin/o2dpg_sim_workflow_anchored.py [hunk 3]
```diff
@@ -329,34 +311,21 @@ def main():
     # make a CCDB accessor object
     ccdbreader = CCDBAccessor(args.ccdb_url)
     # fetch the EOR/SOR
-    rct_sor_eor = retrieve_sor_eor(ccdbreader, args.run_number) # <-- from RCT/Info
-    GLOparams = retrieve_params_fromGRPECS(ccdbreader, args.run_number, rct=rct_sor_eor)
+    run_duration = ccdbreader.get_run_duration(args.run_number)
+    run_start = run_duration.first
+    run_end = run_duration.second
+
+    GLOparams = retrieve_params_fromGRPECS(ccdbreader, args.run_number, run_start=run_start, run_end=run_end)
```
Comment (chiarazampolli) [inline MC/bin/o2dpg_sim_workflow_anchored.py:318]: Is it needed to pass run_start and run_end here?

File: MC/bin/o2dpg_sim_workflow_anchored.py (hunk 1 above)
Comment (benedikt-voelkel) [reply MC/bin/o2dpg_sim_workflow_anchored.py:147]: Yes, as seen by this function, the value from `GRPECS` will be overridden.
We can simply do that silently. Or make the output more clear that all the messages at this stage come from processing the `GRPECS`

**Comment by benedikt-voelkel** [comment]
I see you comments @chiarazampolli and mostly, they seem to go in the direction of "what's printed might be confusing". Maybe there could be some `START processing GRPECS` and `END processing GRPECS`.
Or we simply make those steps silently and do not print anything about it to stdout.

File: MC/bin/o2dpg_sim_workflow_anchored.py (hunk 1 above)
Comment (chiarazampolli) [reply MC/bin/o2dpg_sim_workflow_anchored.py:147]: I understand, but isn't run_start and run_end from RCT? Why GRPECS is used at all to retrieve the SOR?

File: MC/bin/o2dpg_sim_workflow_anchored.py (hunk 3 above)
Comment (chiarazampolli) [reply MC/bin/o2dpg_sim_workflow_anchored.py:318]: Can we simply not avoid to retrieve it from GRPECS now?

**Comment by chiarazampolli** [comment]
Hello @benedikt-voelkel , 
What can be confusing is accessing GRPECS to retrieve start and stop, since now only RCT should be used, right? (apart from overriding, of course).
Chiara

File: MC/bin/o2dpg_sim_workflow_anchored.py (hunk 3 above)
Comment (benedikt-voelkel) [reply MC/bin/o2dpg_sim_workflow_anchored.py:318]: We get the orbit info from that. I asked @shahor02 specifically via email if we should stick to that method and as far as I understand, that should be the way. So we need to process `GRPECS` for that reason.

Comment (chiarazampolli) [reply MC/bin/o2dpg_sim_workflow_anchored.py:318]: Yes, but you can avoid retrieving SOR and EOR, this is what I mean. They will not be consistent with RCT.

Comment (chiarazampolli) [reply MC/bin/o2dpg_sim_workflow_anchored.py:318]: Thanks! This is what I meant.

File: MC/bin/o2dpg_sim_workflow_anchored.py [hunk 4]
```diff
@@ -141,43 +129,25 @@ def retrieve_params_fromGRPECS(ccdbreader, run_number, rct = None):
...
-    if rct != None:
-       # verify that the variaous sor_eor information are the same
-       if SOR != rct["SOR"]:
-         print ("WARNING: Inconsistent SOR information on CCDB (divergence between GRPECS and RCT) ... will take RCT one")
-         SOR=rct["SOR"]
-
-       if EOR != rct["EOR"]:
-         print ("WARNING: Inconsistent EOR information on CCDB (divergence between GRPECS and RCT) ... will take RCT one")
-         EOR=rct["EOR"]
-
```
Comment (benedikt-voelkel) [inline MC/bin/o2dpg_sim_workflow_anchored.py:169]: I took all these things out. Now, simply the given `run_start` and `run_end` are used where needed.

**Comment by benedikt-voelkel** [comment]
@chiarazampolli What do you think? Can we move on?
I left `run_start` and `run_end` for the now called `def retrieve_params_fromGRPECS_and_OrbitReset` function as they are needed for the orbit calculation in there.

**Review by sawenzel — APPROVED** [review]

**Pull request merged**